testing/tests/DevExpress.ui.widgets.pivotGrid/pivotGrid.tests.js

```diff
@@ -2994,6 +2994,126 @@ QUnit.module('dxPivotGrid', {
+
+                const rowsAreaRect = $rowsHeaderArea.get(0).getBoundingClientRect();
+                const columnsAreaRect = $columnsHeaderArea.get(0).getBoundingClientRect();
+
+                const expectedRowCellRect = getHeaderCellByText($rowsHeaderArea, expected.row.toString()).getBoundingClientRect();
+                const expectedColumnCellRect = getHeaderCellByText($columnsHeaderArea, expected.column.toString()).getBoundingClientRect();
+
+                QUnit.assert.roughEqual(rowsAreaRect.top, expectedRowCellRect.top, 16, `expected row position ${message}`);
+                QUnit.assert.roughEqual(columnsAreaRect.left, expectedColumnCellRect.left, 16, `expected column position ${message}`);
+            }
+
+            function tryTriggerScrollEventAndWait(scrollable, clock) {
+                useNative && $(scrollable._container()).trigger('scroll');
+                clock.tick(100);
+            }
+
+            QUnit.test(`PivotGrid -> scrollTo() -> expandHeader -> collapseHeader (T984139). UseNative: ${useNative}, expandDimension: ${area}`, function(assert) {
+                const store = [];
+                for(let i = 0; i < 200; i++) {
+                    store.push({ row: i + 1, column: i + 1, subField: 1, data: 1 });
+                }
+
+                const pivotGrid = createPivotGrid({
+                    width: 1000,
+                    height: 1000,
+                    scrolling: { mode: 'virtual', useNative },
+                    dataSource: {
+                        store: store,
+                        fields: [
+                            { dataField: 'column', area: 'column' },
+                            { dataField: 'row', area: 'row' },
+                            { dataField: 'subField', area: area },
+                            { dataField: 'data', area: 'data' }
+                        ]
+                    }
+                });
+
+                this.clock.tick(100);
+                checkLeftTopVisibleHeaderCellTexts(pivotGrid, { row: '1', column: '1' }, 'after initialization');
+
+                const scrollDistance = browser.msie ? 1950 : 1985; // there is a difference in font size for IE
+                const scrollable = pivotGrid._dataArea.groupElement().dxScrollable('instance');
+                scrollable.scrollTo({ left: scrollDistance, top: 2000 });
+                tryTriggerScrollEventAndWait(scrollable, this.clock);
+
+                const expectedRow = 60;
+                const expectedColumn = area === 'row' ? 57 : 46;
+                checkLeftTopVisibleHeaderCellTexts(pivotGrid, { row: expectedRow, column: expectedColumn }, 'after scrolling');
+
+                const nodeToExpand = 65;
+                const getExpandedCells = () => pivotGrid.$element().find('.dx-pivotgrid-expanded');
+
+                pivotGrid.getDataSource().expandHeaderItem(area, [nodeToExpand]);
+                tryTriggerScrollEventAndWait(scrollable, this.clock);
```

Comment:
Why test code should manually trigger 'scroll' after 'expandHeaderItem'?
Can test code call a method that executes all the necessary code?